dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/model/ProviderMethodModel.java

```diff
@@ -17,17 +17,18 @@
 package org.apache.dubbo.rpc.model;
 
 import java.lang.reflect.Method;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 public class ProviderMethodModel {
-    private transient final Method method;
+    private final Method method;
     private final String methodName;
     private final String[] methodArgTypes;
-    private final String serviceName;
 
+    private final ConcurrentMap<String, Object> attributeMap = new ConcurrentHashMap<>();
```

Comment:
`private final Map<String, Object> attributeMap = new ConcurrentHashMap<>();` would be better.